account_invoice_alternate_payer/models/account_move.py

```diff
@@ -0,0 +1,161 @@
+        )._compute_bank_partner_id()
+
+    @api.onchange("alternate_payer_id")
+    def _onchange_alternate_payer_id(self):
+        return self._onchange_partner_id()
+
+    @contextmanager
+    def _sync_dynamic_line(
+        self,
+        existing_key_fname,
+        needed_vals_fname,
+        needed_dirty_fname,
+        line_type,
+        container,
+    ):
+        with super()._sync_dynamic_line(
+            existing_key_fname,
+            needed_vals_fname,
+            needed_dirty_fname,
+            line_type,
+            container,
+        ):
+            if line_type == "payment_term" and container.get("records").filtered(
+                lambda x: x.alternate_payer_id
+            ):
+                for invoice in container.get("records").filtered(
+                    lambda x: x.alternate_payer_id
+                ):
+                    payment_term_lines = invoice.line_ids.filtered(
+                        lambda x: x.display_type == "payment_term"
+                    )
+                    payment_term_lines.write(
+                        {
+                            "partner_id": invoice.alternate_payer_id.id,
+                        }
+                    )
+            yield
+
+    def _compute_payments_widget_to_reconcile_info(self):
+        super_moves = self.filtered(lambda r: not r.alternate_payer_id)
+        for move in self - super_moves:
+            move.invoice_outstanding_credits_debits_widget = False
+            move.invoice_has_outstanding = False
+            if (
+                move.state != "posted"
+                or move.payment_state != "not_paid"
+                or not move.is_invoice(include_receipts=True)
+            ):
+                continue
+            pay_term_line_ids = move.line_ids.filtered(
+                lambda line: line.account_id.account_type
+                in ("asset_receivable", "liability_payable")
+            )
+
+            domain = [
+                ("account_id", "in", pay_term_line_ids.mapped("account_id").ids),
+                "|",
+                ("move_id.state", "=", "posted"),
+                "&",
+                ("move_id.state", "=", "draft"),
+                ("partner_id", "=", move.alternate_payer_id.id),
+                ("reconciled", "=", False),
+                "|",
+                ("amount_residual", "!=", 0.0),
+                ("amount_residual_currency", "!=", 0.0),
+            ]
+
+            if move.is_inbound():
+                domain.extend([("credit", ">", 0), ("debit", "=", 0)])
+                type_payment = _("Outstanding credits")
+            else:
+                domain.extend([("credit", "=", 0), ("debit", ">", 0)])
+                type_payment = _("Outstanding debits")
+            info = {"title": "", "outstanding": True, "content": [], "move_id": move.id}
+            lines = self.env["account.move.line"].search(domain)
+            currency_id = move.currency_id
+            if len(lines) != 0:
```

Comment:
Innecessary `if` . In case `lines` is empty, it will not enter the `for` loop.

Reply:
Change to loop over for only